include/forcing/Forcing.h

```diff
@@ -257,46 +356,61 @@ class Forcing
+                    AORC.VGRD_10maboveground_meters_per_second = atof(VGRD_10maboveground_str.c_str());
+                  
+                    //Add AORC struct to AORC vector
+                    AORC_vector.push_back(AORC);
+            
+                    //Convert precip_rate from string to double
                     double precip_rate = atof(precip_rate_str.c_str());
 
                     //Add data to vectors
-                    APCP_surface_kg_per_meters_squared_vector.push_back(APCP_surface);
-                    DLWRF_surface_W_per_meters_squared_vector.push_back(DLWRF_surface);
-                    DSWRF_surface_W_per_meters_squared_vector.push_back(DSWRF_surface);
-                    PRES_surface_Pa_vector.push_back(PRES_surface);
-                    SPFH_2maboveground_kg_per_kg_vector.push_back(SPFH_2maboveground);
-                    TMP_2maboveground_K_vector.push_back(TMP_2maboveground);
-                    UGRD_10maboveground_meters_per_second_vector.push_back(UGRD_10maboveground);
-                    VGRD_10maboveground_meters_per_second_vector.push_back(VGRD_10maboveground);
                     precipitation_rate_meters_per_second_vector.push_back(precip_rate);
+                    time_epoch_vector.push_back(current_row_date_time_epoch);
                 }
 
                 //Free memory from struct
-                //delete current_row_date_time;
+                //delete current_row_date_time_utc;
         }
+
+        //Get size of forcing precipitation rate vector
+        forcing_vector_size = precipitation_rate_meters_per_second_vector.size();
     }
 
-    vector<double> APCP_surface_kg_per_meters_squared_vector;
-    vector<double> DLWRF_surface_W_per_meters_squared_vector;
-    vector<double> DSWRF_surface_W_per_meters_squared_vector;
-    vector<double> PRES_surface_Pa_vector;
-    vector<double> SPFH_2maboveground_kg_per_kg_vector;
-    vector<double> TMP_2maboveground_K_vector;
-    vector<double> UGRD_10maboveground_meters_per_second_vector;
-    vector<double> VGRD_10maboveground_meters_per_second_vector;
+    //AORC Forcing Data Struct
+    struct AORC_data
+    {
+        double APCP_surface_kg_per_meters_squared; //Total Precipitation (kg/m^2)
+        double DLWRF_surface_W_per_meters_squared; //Downward Long-Wave Rad. (Flux W/m^2)
+        double DSWRF_surface_W_per_meters_squared; //Downward Short-Wave Radiation (Flux W/m^2)
+        double PRES_surface_Pa; //Pressure (Pa)
+        double SPFH_2maboveground_kg_per_kg; //Specific Humidity (kg/kg)
+        double TMP_2maboveground_K; //Temperature (K)
+        double UGRD_10maboveground_meters_per_second; //U-Component of Wind (m/s)
+        double VGRD_10maboveground_meters_per_second; //V-Component of Wind (m/s)
+    };
+
+    vector<AORC_data> AORC_vector;
+
     vector<double> precipitation_rate_meters_per_second_vector;
+    vector<time_t> time_epoch_vector; 
     int forcing_vector_index_ptr;
+    int forcing_vector_size;
```

Comment:
This is tracked in the vector `precipitation_rate_meters_per_second_vector.size()`
It is probably better to use this anywhere you need size in case future work allows this vector to change size during the lifetime of a forcing object.

Reply:
Ok thanks. I will replace that.